torch/csrc/generic/StorageSharing.cpp

```diff
@@ -42,12 +42,17 @@ static PyObject * THPStorage_(sharedIncref)(PyObject *_self, PyObject *noargs)
 
 #ifndef THC_GENERIC_FILE
 
-static THWStorage* THPStorage_(newFilenameStorage)(ptrdiff_t size)
+static c10::StorageImpl* THPStorage_(newFilenameStorage)(ptrdiff_t size)
 {
   int flags = at::ALLOCATOR_MAPPED_SHAREDMEM | at::ALLOCATOR_MAPPED_EXCLUSIVE;
   std::string handle = at::NewProcessWideShmHandle();
-  return THWStorage_(newWithDataAndAllocator)(
-      THManagedMapAllocator::makeDataPtr("", handle.c_str(), flags, size * sizeof(scalar_t)), size, /* allocator */ nullptr);
+  return c10::make_intrusive<at::StorageImpl>(
```

Comment:
given how many times `newWithDataAndAllocator` is used, maybe makes sense to factor it out?

Reply:
I assume you mean `newFilenameStorage`, not `newWithDataAndAllocator`. Yeah it seems to be only called twice, so I'll remove it